### PR TITLE
Issue9 create auth

### PR DESCRIPTION
**Pull request by almightyboz**
* Add Devise to gemfile, configure env/development.rb
* Create users table, add names and roles
* Add Devise views

### CODE REVIEW

**Review by jmkoni — APPROVED** [review]
👍

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -9,6 +9,8 @@
   </head>
 
   <body>
+	<p class="notice"><%= notice %></p>
```
Comment (osondoar) [inline app/views/layouts/application.html.erb:12]: `application.html.erb` is going to probably be the layout for the public site. The flash message should only be needed in the layout for the internal site (which we don't have yet). I think we can remove those flash messages for now.

File: config/environments/development.rb [hunk 2]
```diff
@@ -51,4 +51,7 @@
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
   config.file_watcher = ActiveSupport::EventedFileUpdateChecker
+  
+  # TODO: set :host in environments/production.rb to be the domain of our production app
+  config.action_mailer.default_url_options = { :host => 'justice.dev' }
```
Comment (osondoar) [inline config/environments/development.rb:56]: The host refers to the mail server. I would leave it to `config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }` in development.

Btw, this tool is great for testing emails locally: https://mailcatcher.me/

File: test/models/user_test.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (osondoar) [inline test/models/user_test.rb:1]: This looks like a mini test spec. Can you make it a rspec one?

**Comment by osondoar** [comment]
Just added a couple comments, but 👍 overall.

**Review by iamanissa — COMMENTED** [review]
for the schema, do we want usernames, or just name?

**Review by jmkoni — CHANGES_REQUESTED** [review]
Right now, there are also a lot of commits in this PR that aren't yours. Usually that means a rebase went haywire.

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (almightyboz) [reply app/views/layouts/application.html.erb:12]: A note, Devise requested that we have those in for flash messages after login/logout. I'll put them in the internal site when we have it.

**Pull request closed without merging**